src/tests/EShopworld.WorkerProcess.IntegrationTests/WorkerLeaseTests.cs

```diff
@@ -126,6 +126,40 @@ public async Task TestMultipleWorkLeases()
             _expiredList.Should().NotBeEmpty();
         }
 
+        [Fact, IsIntegration]
+        public async Task TestMultipleConcurrentWorkLeases()
+        {
+            ManualResetEvent manualResetEvent = new ManualResetEvent(false);
+
+            // Arrange
+            Random r = new Random();
+            var leaseStore = _serviceProvider.GetService<ILeaseStore>();
+            SetupWorkerLeases(30, i => r.Next(7));
+
+            // Act
+            await leaseStore.InitialiseAsync();
+
+            foreach (var workerLease in _workerLeases)
+            {
+                workerLease.StartLeasing();
+
+                _output.WriteLine($"[{DateTime.UtcNow}] Starting [{workerLease.InstanceId}]");
+            }
+
+            manualResetEvent.WaitOne(new TimeSpan(0, 1, 30));
+
+            foreach (var workerLease in _workerLeases)
+            {
+                workerLease.StopLeasing();
+
+                _output.WriteLine($"Stopping [{workerLease.InstanceId}]");
+            }
+
+            // Assert
+            _allocatedList.Select(w=>w.InstanceId).Count().Should().Be(1);
+            _allocatedList.First().Priority.Should().Be(_workerLeases.Select(w => w.Priority).Min());
```

Comment:
Very-very small comment: you could have a assertion scope here.

Reply:
sure it's done